### PR TITLE
merged changes from original ehrbase feature/264_name_of_item_tree fo…

### PR DESCRIPTION
**Pull request by chevalleyc**
…llowing project restructuring

### CODE REVIEW

**Comment by chevalleyc** [comment]
This is a refactoring for https://github.com/ehrbase/ehrbase/pull/246 (see this PR for more details)

File: serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvCodedTextNameValue.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    /**
+     * Encode a name value into the DB json structure
+     * <code>
+     *     "name": {
+     *         "value":...
+     *     }
+     * </code>
+     * @throws IOException
+     */
+    public void write() throws IOException {
```
Comment (jakesmolka) [inline serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvCodedTextNameValue.java:56]: Add `@Override`?

File: serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvTextNameValue.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * Encode a name value into the DB json structure
+     * <code>
+     *     "name": {
+     *         "value":...
+     *     }
+     * </code>
+     * @throws IOException
+     */
+    public void write() throws IOException {
```
Comment (jakesmolka) [inline serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvTextNameValue.java:51]: Add `@Override`?

File: serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvCodedTextNameValue.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.Map;
+
+public class DvCodedTextNameValue implements I_NameValueHandler{
+
+    private final JsonWriter writer;
+    private final String value;
+    private final String codeString;
+    private final String terminologyId;
+
+    DvCodedTextNameValue(JsonWriter writer, String value) {
```
Comment (jakesmolka) [inline serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvCodedTextNameValue.java:35]: Not used nor necessary by interface. So, safe to delete?

File: serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvCodedTextNameValue.java (hunk 1 above)
Comment (chevalleyc) [reply serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvCodedTextNameValue.java:53]: Agreed. Makes the code neater

File: serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvTextNameValue.java (hunk 2 above)
Comment (chevalleyc) [reply serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvTextNameValue.java:52]: Agreed. Makes the code neater

File: serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvCodedTextNameValue.java (hunk 3 above)
Comment (chevalleyc) [reply serialisation/src/main/java/org/ehrbase/serialisation/dbencoding/wrappers/json/writer/translator_db2raw/DvCodedTextNameValue.java]: Yep. Done

**Pull request merged**